fix account number switch macro
fix account deletion
publish gpg using gpg-wks protocol
dynamically load mailboxes
use /bin/sh instead of /bin/bash, remove non-POSIX PWR signal

Index: bin/mw
--- bin/mw.orig
+++ bin/mw
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh
 
 set -a
 
@@ -152,7 +152,7 @@
 	sed -ibu "/IMAPStore $fulladdr-remote$/,/# End profile/d" "$mbsyncrc" 2>/dev/null
 	rm -f "$mbsyncrc"bu
 	rm -rf "${cachedir:?}/${fulladdr:?}" "$accdir/$fulladdr.muttrc" "$accdir/"[0-9]-"$fulladdr.muttrc"
-	sed -ibu "/\([0-9]-\)\?$fulladdr.muttrc/d" "$muttrc" 2>/dev/null
+	sed -ibu "/macro.*i[0-9].*$fulladdr.muttrc/d" "$muttrc" 2>/dev/null
 	rm -f "$muttrc"bu
 	sed -ibu "/account $fulladdr$/,/^\(\s*$\|account\)/d" "$msmtprc" 2>/dev/null
 	rm -f "$msmtprc"bu
@@ -225,14 +225,11 @@
 		mailboxes="$(echo "$info" | ggrep -v HasChildren | sed "s/.*\" //;s/\"//g" | tr -d '\r')"
 	fi
 	[ "$type" = "pop" ] && mailboxes="INBOX"
-	for x in $(
-		sed -n "/^macro.* i[0-9] / s/\(^macro.* i\| .*\)//gp " "$muttrc" 2>/dev/null | sort -u
-		echo 0
-	); do
+	for x in $(ggrep -Eo "^macro.* i[0-9] " "$muttrc" | sed "s/^macro.* i//g" | sort -u; echo 0); do
 		idnum=$((idnum + 1))
 		[ "$idnum" -eq "$x" ] || break
 	done
-	toappend="mailboxes $(echo "$mailboxes" | sed "s/^/\"=/;s/$/\"/;s/'/\\\'/g" | paste -sd ' ' -)"
+	toappend="mailboxes \`mdir=$maildir/$fulladdr/; find \$mdir -mindepth 1 -type d -name cur | sed -e 's:/cur\$:\":' -e \"s:\$mdir:\\\"=:\" | sort | tr '\\\n' ' '\`"
 }
 
 finalize() {
@@ -311,14 +308,14 @@
 
 reorder() {
 	tempfile="$(mktemp -u)"
-	trap 'rm -f $tempfile' HUP INT QUIT TERM PWR EXIT
+	trap 'rm -f $tempfile' HUP INT QUIT TERM EXIT
 	echo "# Carefully reorder these accounts with the desired numbers in the first column.
 # DO NOT reorder rows or rename the accounts in the second column." >"$tempfile"
 	sed -n "
 	/ i[0-9] / s?\(.* i\|'<sync.*/\|\.muttrc.*\)??g p
 	" "$muttrc" >>"$tempfile"
 	${EDITOR:-vim} "$tempfile" || exit 1
-	sed -i -e 's/#.*//' -e '/^$/d' "$tempfile"
+	sed -i '' -e 's/#.*//' -e '/^$/d' "$tempfile"
 	default="$(sort -n "$tempfile" | head -n 1)"
 	default="${default#* }"
 	sed -ibu "
